castai/resource_castai_cluster.go

```diff
@@ -217,7 +399,101 @@ func waitForClusterToReachStatusFunc(ctx context.Context, client *sdk.ClientWith
 				return resource.RetryableError(fmt.Errorf("waiting for cluster to reach %q status, id=%q name=%q, status=%s", targetStatus, cluster.Id, cluster.Name, cluster.Status))
 			}
 		}
-
 		return resource.NonRetryableError(fmt.Errorf("cluster has reached unexpected status, id=%q name=%q, status=%s", cluster.Id, cluster.Name, cluster.Status))
 	}
 }
+
+func expandCPUlimit(params interface{}) sdk.ClusterLimitsCpu {
+	var clusterLimitsCPU sdk.ClusterLimitsCpu
+	for _, val := range params.([]interface{}) {
+		cpuData := val.(map[string]interface{})
+		clusterLimitsCPU = sdk.ClusterLimitsCpu{
+			MaxCores:  int64(cpuData["max_cores"].(int)),
+			MinCores:  int64(cpuData["min_cores"].(int)),
+		}
+	}
+	return clusterLimitsCPU
+}
+
+func expandNodeDownscaler(params interface{}) *sdk.NodeDownscalerEmptyNodes {
+	var nodeDownscalerBool sdk.NodeDownscalerEmptyNodes
+	for _, val := range params.([]interface{}) {
+		ndData := val.(map[string]interface{})
+		nodeDownscalerEnabled := ndData["enabled"].(bool)
+		nodeDownscalerBool = sdk.NodeDownscalerEmptyNodes{
+			Enabled:  &nodeDownscalerEnabled,
+		}
+	}
+	return &nodeDownscalerBool
+}
+
+func expandHeadroom(params interface{}) sdk.Headroom {
+	var headroomPercentage sdk.Headroom
+	for _, val := range params.([]interface{}) {
+		hpData := val.(map[string]interface{})
+		headroomPercentage = sdk.Headroom{
+			CpuPercentage:  	hpData["cpu_percentage"].(int),
+			MemoryPercentage:   hpData["memory_percentage"].(int),
+		}
+	}
+	return headroomPercentage
+}
+
+func updatePolicies(ctx context.Context, client *sdk.ClientWithResponses, clusterID string, pc map[string]interface{}) diag.Diagnostics {
+	var clusterLimits sdk.ClusterLimitsPolicy
+	for _, val := range pc[PolicyFieldClusterLimits].([]interface{}) {
+		limitData := val.(map[string]interface{})
+
+		clusterLimits = sdk.ClusterLimitsPolicy{
+			Enabled:  limitData["enabled"].(bool),
+			Cpu: expandCPUlimit(limitData["cpu"].(interface{})),
+		}
+	}
+	log.Printf("[DEBUG] Reading Policies `cluster_limits` parameterEnabled=%v", clusterLimits)
```

Comment:
`parameterEnabled=%v` is this correct here? 

My suggestion is to have one expansion function for entire policies, then updatePolicies becomes:
1. Expand the policies 
2. Print them for logging
3. Send them to api